src/Parser/Ast/FullClassName.php

```diff
@@ -6,19 +6,22 @@ class FullClassName implements ClassLike
 {
     private $namespace;
     private $name;
+    private $fqcn;
 
-    public function __construct(string $namespace, string $name)
+    private function __construct(string $namespace, string $name, string $fqcn)
     {
         $this->namespace = $namespace;
         $this->name = $name;
+        $this->fqcn = $fqcn;
     }
 
     public static function createFromFQCN(string $fqcn): self
     {
         $parts = explode('\\', ltrim($fqcn, '\\'));
         $name = array_pop($parts);
+        $normalizedFqcn = empty($parts) ? $name : $fqcn;
```

Comment:
This ternary remove the initial slash from classes at the root namespace. It is functionnaly equivalent to the one removed below from `getFQCN`.